x/evm/keeper/params.go

```diff
@@ -49,25 +49,55 @@ func (k Keeper) GetLegacyParams(ctx sdk.Context) types.Params {
 	return params
 }
 
-// EnablePrecompiles appends the addresses of the given Precompiles to the list
+// EnableStaticPrecompiles appends the addresses of the given Precompiles to the list
 // of active precompiles.
-func (k Keeper) EnablePrecompiles(ctx sdk.Context, addresses ...common.Address) error {
+func (k Keeper) EnableStaticPrecompiles(ctx sdk.Context, addresses ...common.Address) error {
 	params := k.GetParams(ctx)
 	activePrecompiles := params.ActivePrecompiles
 
-	for _, address := range addresses {
-		activePrecompiles = append(activePrecompiles, address.String())
-	}
-
-	sort.Slice(activePrecompiles, func(i, j int) bool {
-		return activePrecompiles[i] < activePrecompiles[j]
-	})
+	// Append and sort the new precompiles
+	activePrecompiles = appendPrecompiles(activePrecompiles, addresses...)
 
 	params.ActivePrecompiles = activePrecompiles
+	return k.SetParams(ctx, params)
+}
+
+// EnableDynamicPrecompiles appends the addresses of the given Precompiles to the list
+// of active precompiles.
+func (k Keeper) EnableDynamicPrecompiles(ctx sdk.Context, addresses ...common.Address) error {
+	// Get the current params and append the new precompiles
+	params := k.GetParams(ctx)
+	activePrecompiles := params.ActiveDynamicPrecompiles
 
+	// Append and sort the new precompiles
+	activePrecompiles = appendPrecompiles(activePrecompiles, addresses...)
+
+	// Update params
+	params.ActiveDynamicPrecompiles = activePrecompiles
 	return k.SetParams(ctx, params)
 }
 
+func appendPrecompiles(existingPrecompiles []string, addresses ...common.Address) []string {
+	updatedPrecompiles := []string{}
+	for _, address := range addresses {
+		// Check for duplicates
+		if !slices.Contains(existingPrecompiles, address.String()) {
+			// return nil, fmt.Errorf("precompile already registered: %s", address)
+			updatedPrecompiles = append(updatedPrecompiles, address.String())
```

Comment:
how come we don't want the duplicates check anymore? We can remove the error line if unnecessary 👍

Reply:
I dont think this was an intended change